### ISSUE TITLE
Unable to authenticate from S3Proxy

### ISSUE THREAD

**Issue by larshagencognite** [Azure/Azurite#2233]
### Which service(blob, file, queue, table) does this issue concern?

blob

### Which version of the Azurite was used?

3.25.0

### Where do you get Azurite? (npm, DockerHub, NuGet, Visual Studio Code Extension)

DockerHub

### What's the Node.js version?

v14.21.3

### What problem was encountered?

Unable to authenticate with azurite from S3Proxy (using SharedKeyLite authorization)
https://github.com/gaul/s3proxy

S3Proxy uses JClouds, which in turn uses SharedKeyLite authorization, which is currently not supported by Azurite. Supporting this would be great, as it would allow local testing and development using S3Proxy and Azurite.

### Steps to reproduce the issue?

Run azurite docker image
```
docker run -d --name azurite -p 10000:10000 -p 10001:10001 mcr.microsoft.com/azure-storage/azurite:3.25.0 azurite --blobHost 0.0.0.0 --queueHost 0.0.0.0 --loose -d /dev/stdout
```
Run S3Proxy with the following config:
```
s3proxy.endpoint=http://127.0.0.1:0
s3proxy.authorization=aws-v2
s3proxy.identity=local-identity
s3proxy.credential=local-credential

jclouds.provider=azureblob
jclouds.azureblob.auth=azureKey
jclouds.endpoint=http://127.0.0.1:10000/devstoreaccount1
jclouds.identity=devstoreaccount1
jclouds.credential=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==
```

Run a put request against S3Proxy to create a container

In [SharedKeyLiteAuthentication.java](https://github.com/apache/jclouds/blob/master/providers/azureblob/src/main/java/org/jclouds/azure/storage/filters/SharedKeyLiteAuthentication.java), S3Proxy will sign string similar to this:
```
PUT


Wed, 18 Oct 2023 07:58:29 GMT
x-ms-version:2017-11-09
/devstoreaccount1/devstoreaccount1/s3proxy-1306537511
```

The logs from Azurite will include something like:
```
BlobSharedKeyAuthenticator:validate() Start validation against account shared key authentication.
BlobSharedKeyAuthenticator:validate() [STRING TO SIGN]:"PUT\n\n\n\n\n\nTue, 17 Oct 2023 13:25:53 GMT\n\n\n\n\n\nx-ms-version:2017-11-09\n/devstoreaccount1/devstoreaccount1/s3proxy-1005549727\nrestype:container"
BlobSharedKeyAuthenticator:validate() Calculated authentication header based on key1: SharedKey devstoreaccount1:a3dTuRHqQl0TqIkzog0LS4gxm4ViqQPkKyUztEgM0qo=
BlobSharedKeyAuthenticator:validate() Validation failed.
```
The string it expects to be signed is:
```
PUT





Tue, 17 Oct 2023 13:25:53 GMT





x-ms-version:2017-11-09
/devstoreaccount1/devstoreaccount1/s3proxy-1005549727
restype:container
```

I assume this is just the difference between the SharedKey and SharedKeyLite scheme.

### Have you found a mitigation/solution?

no

**Comment by larshagencognite** [comment]
This would be fixed by https://github.com/Azure/Azurite/pull/1752

Duplicate of https://github.com/Azure/Azurite/issues/378

**Comment by blueww** [comment]
@larshagencognite 

Thanks for raising this issue and find the existing same issue!

The before PR https://github.com/Azure/Azurite/pull/1752 has pending for a long time, for missing test case.

Azurite welcome contribution!
It would be great if you could raise a PR to add the feature, or continue the before PR work to add test case and fix conflict, and then we can continue to review/merge it.

**Comment by larshagencognite** [comment]
As I am unfamiliar with typescript, and SharedKey is the newer standard, I have opted to unblock our use case by implementing SharedKey support in jclouds instead https://github.com/apache/jclouds/pull/186

**Comment by blueww** [comment]
@larshagencognite 

Thanks for the update!
As there's another issue https://github.com/Azure/Azurite/issues/378 and PR https://github.com/Azure/Azurite/pull/1752 to track the SharedkeyLite support, I will close this issue soon.
Feel free to let us know if you have any concern.
